webui/react/src/services/decoder.ts

```diff
@@ -460,29 +462,67 @@ export const mapSearchExperiment = (
 ): types.ExperimentWithTrial => {
   return {
     bestTrial: data.bestTrial && decodeV1TrialToTrialItem(data.bestTrial),
-    experiment: data.experiment && mapV1Experiment(data.experiment),
+    experiment: data.experiment && (mapV1Experiment(data.experiment) as types.BulkExperimentItem),
   };
 };
 
 export const mapV1Experiment = (
   data: Sdk.V1Experiment,
   jobSummary?: types.JobSummary,
-): types.ExperimentItem => {
+  config?: Sdk.V1GetExperimentResponse['config'],
+): types.FullExperimentItem | types.BulkExperimentItem => {
```

Comment:
I cannot do conditional type based on the function argument like `config === undefined ? types.BulkExperimentItem : types.FullExperimentItem`, so i used `as` to assert type around where `mapV1Experiment` is called, but this should be safe

Reply:
you can, you just need to overload the function definition
```tsx
export function mapV1Experiment (data: Sdk.V1Experiment, config?: undefined, jobSummary?: types.JobSummary): types.BulkExperimentItem
export function mapV1Experiment (data: Sdk.V1Experiment, config: Sdk.V1GetExperimentResponse['config'], jobSummary?: types.JobSummary): types.FullExperimentItem
export function mapV1Experiment (data: SdkV1Experiment, config?: Sdk.V1GetExperimentResponse['config'], jobSummary?: types.JobSummary): types.BulkExperimentItem | types.fullExperimentItem {
// function body here
```

Reply:
so true, forgot typescript support overload. thanks